native/src/ripper_tree_types.rs

```diff
@@ -474,42 +474,11 @@ pub enum StringContentPart {
 }
 
 def_tag!(string_content_tag, "string_content");
-#[derive(Debug, Clone)]
-pub struct StringContent(pub string_content_tag, pub Vec<StringContentPart>);
-
-impl<'de> Deserialize<'de> for StringContent {
-    fn deserialize<D>(deserializer: D) -> Result<StringContent, D::Error>
-    where
-        D: Deserializer<'de>,
-    {
-        struct StringContentVisitor;
-
-        impl<'de> de::Visitor<'de> for StringContentVisitor {
-            type Value = StringContent;
-
-            fn expecting(&self, f: &mut std::fmt::Formatter<'_>) -> Result<(), std::fmt::Error> {
-                write!(
-                    f,
-                    "[string_content, (tstring_content, string_embexpr, string_dvar)*]"
-                )
-            }
-
-            fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
-            where
-                A: de::SeqAccess<'de>,
-            {
-                let tag = seq
-                    .next_element()?
-                    .ok_or_else(|| de::Error::invalid_length(0, &self))?;
-                let elements =
-                    Deserialize::deserialize(de::value::SeqAccessDeserializer::new(&mut seq))?;
-                Ok(StringContent(tag, elements))
-            }
-        }
-
-        deserializer.deserialize_seq(StringContentVisitor)
-    }
-}
+#[derive(Deserialize, Debug, Clone)]
+pub struct StringContent(
+    pub string_content_tag,
+    #[serde(flatten)] pub Vec<StringContentPart>,
```

Comment:
can the flatten annotations go on the line above, as opposed to the same line?

Reply:
rustfmt says no

Reply:
I want to remove the tags so this will eventually be on a single line anyway. And/or I'll change it to a named struct in which case it'll go on its own line